ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx

```diff
@@ -130,122 +101,119 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
     getByDisplayValue("Prop");
   });
 
-  it("renders child rule group", () => {
-    const property: PropertyDescription = {
-      displayLabel: "Prop",
-      name: "prop",
-      typename: "int",
-    };
-    const { getByText } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer
-        {...defaultProps}
-        group={{
-          id: "id",
-          operator: PropertyFilterRuleGroupOperator.And,
-          items: [
-            {
-              id: "childId",
-              groupId: "id",
-              operator: PropertyFilterRuleGroupOperator.Or,
-              items: [
-                {
-                  id: "grandChildId1",
-                  groupId: "childId",
-                  property,
-                },
-                {
-                  id: "grandChildId2",
-                  groupId: "childId",
-                  property,
-                },
-              ],
-            },
-          ],
-        }}
-      />
-    );
-
-    getByText(
-      TestUtils.i18n.getLocalizedString("Components:filterBuilder.operators.or")
-    );
-  });
-
-  it("dispatches add rule event when button is clicked", () => {
+  it("dispatches operator change event when operator is selected", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
+    const { container, findByText } = renderWithContext(
       <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
       { actions }
     );
-    const addItemSpy = sinon.stub(actions, "addItem");
+    const setRuleGroupOperatorSpy = sinon.stub(actions, "setRuleGroupOperator");
 
-    const addRuleButton = getByTestId("rule-group-add-rule");
-    addRuleButton.click();
+    const selector = container.querySelector<HTMLAnchorElement>(
+      ".fb-group-operator .fb-logical-operator-toggle"
+    );
+    expect(selector).to.not.be.null;
+
+    selector?.click();
+
+    expect(
+      await findByText(
+        TestUtils.i18n.getLocalizedString(
+          "Components:filterBuilder.operators.and"
+        )
+      )
+    ).to.not.be.null;
 
-    expect(addItemSpy).to.be.calledOnceWith(defaultProps.path, "RULE");
+    expect(setRuleGroupOperatorSpy).to.be.calledWith(
+      defaultProps.path,
+      PropertyFilterRuleGroupOperator.Or
+    );
   });
 
-  it("dispatches add rule group event when button is clicked", () => {
+  it("Toggles operator 'Or' to 'And'", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroup,
```

Comment:
On the other PR, there was a remark about mutating `rootGroup`, I think it was missed in the branch.
```suggestion
    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
    const props: PropertyFilterBuilderRuleGroupRendererProps = {
      group: { ...rootGroup, operator: PropertyFilterRuleGroupOperator.Or },
```